plugin/core/sessions.py

```diff
@@ -220,24 +275,49 @@ def clear_dotted_value(current: dict, dotted: str) -> None:
     current.pop(keys[-1], None)
 
 
-class Session(object):
-    def __init__(self,
-                 config: ClientConfig,
-                 workspace_folders: List[WorkspaceFolder],
-                 client: Client,
-                 on_pre_initialize: 'Optional[Callable[[Session], None]]' = None,
-                 on_post_initialize: 'Optional[Callable[[Session], None]]' = None,
-                 on_post_exit: Optional[Callable[[str], None]] = None) -> None:
+class Session(Client):
+
+    @classmethod
+    def name(cls) -> str:
+        raise NotImplementedError()
+
+    @classmethod
+    def needs_update_or_installation(cls) -> bool:
+        return False
+
+    @classmethod
+    def install_or_update(cls) -> None:
+        pass
+
+    @classmethod
+    def standard_configuration(cls) -> ClientConfig:
```

Comment:
**standard**_configuration as opposed to what? Wouldn't it be enough to call it `configuration`?